crates/common/src/dotrain_order/mod.rs

```diff
@@ -427,13 +466,30 @@ impl DotrainOrder {
 
         order.inputs = new_inputs;
         order.outputs = new_outputs;
-        deployment.order = Arc::new(order);
-        self.config
-            .deployments
-            .insert(deployment_name.to_string(), Arc::new(deployment));
+        self.config_source
+            .orders
+            .insert(deployment.order.clone(), order.clone());
+        self.update_config_source(self.config_source.clone())?;
 
         Ok(())
     }
+
+    pub fn update_token_address(
```

Comment:
should this also update token infos afterwards?

Reply:
token information is updated in the gui library so it happens here https://github.com/rainlanguage/rain.orderbook/blob/44deaafa0d6b5d1ef921b95750e1725cff8ba045/crates/js_api/src/gui/select_tokens.rs#L53